java/src/jmri/jmrit/logix/WarrantShutdownTask.java

```diff
@@ -0,0 +1,107 @@
+package jmri.jmrit.logix;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import jmri.InstanceManager;
+import jmri.implementation.AbstractShutDownTask;
+import jmri.jmrit.roster.Roster;
+import jmri.jmrit.roster.RosterEntry;
+import jmri.jmrit.roster.RosterSpeedProfile;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * 
+ * Allows user to decide if (and which) SpeedProfiles to write to the Roster at 
+ * the end of a session.  Locos running warrants have had their speeds measured
+ * and this new data may or may not be merged into any existing SpeedProfiles
+ * in the Roster.
+ * <p>
+ *
+ * @author Pete cressman Copyright (C) 2017
+ */
+public class WarrantShutdownTask extends AbstractShutDownTask {
```

Comment:
Note that if you want to allow user intervention in the shutdown process, you should extend SwingShutDownTask instead of AbstractShutDownTask.